### ISSUE TITLE
Pipes and wires/possibly related to #88?

### ISSUE THREAD

**Issue by 34486** [GregTechCEu/GregTechCEu-1.19#102]
### GregTech CEu Version

1.0.4

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

Yes

### Expected Behavior

Machines to start operation(s) when items and required energy source supplied, along with the pipes and cables to stay connected to transport energy or fluids. 

### Actual Behavior

I am having issues possibly related to issue #88 where I have a steam macerator or any other steam machine, and I can give it items to process, it will complete the items that I gave it but if I remove them or later add more to process, I must break the machine and place it again to get it to work. Leaving and joining the singleplayer world again does not solve the issue. 

Also, when I leave the world and join again, all pipes and power cables are visibly connected but do not transport whatever they are supposed to transport, such as water, steam or power. I must break all cables/pipes and replace all of them to reconnect them.  If I just break one cable or pipe and place it back, it will refuse to connect even when using a wrench on it to manually disconnect/connect it. 

### Steps to Reproduce

Set up a boiler fed by water from some pipes, hook up any steam machine using pipes again, start with a macerator. Put ore in it to process, let it complete the task. Try to place more ore into it, it will not process. Break and replace the machine and its fixed, though it voids steam it has internally stored. Now, leave the world and rejoin it, try to process ore. It will visibly be connected to pipes but they will not supply with anything. Break and place all pipes again and it will work.

### Additional Information

![2023-06-18_10 05 18](https://github.com/GregTechCEu/GregTechCEu-1.19/assets/19638851/cbe138e6-97b5-4042-9b61-9c56ee8e5965)

Not too sure how screenshots might help but there is one of my setup. I must replace all fluid pipes to get them to do their thing and after each full task is completed by a machine (other than steam boiler) I must break and replace the machine for it to be able to operate again.

**Comment by lukaslcf** [comment]
Are you playing on fabric or forge?

**Comment by lukaslcf** [comment]
Btw in the image, it seems like the muffler at the back of the machine is blocked by the steam pipe, can you double check?

**Comment by 34486** [comment]
Forge. Oh yes though that is part of the issue, odd that it will process a stack of items without having to let steam out the back. I was confused for a while as the texture is glitching even though I rotated the machine and never saw one it eventually showed up. So I guess it does operations without letting steam out which is a bug I assume..? 

Still the pipes and wires "disconnect" when you relog from world. They visibly are connected but do not do anything unless all broken and replaced.

**Comment by lukaslcf** [comment]
It can do one operation without letting steam out. But doing a stack is indeed probably a bug...

I still don't see the "disconnect" that you mention though

**Comment by 34486** [comment]
I can set up a boiler fed by wooden pipe, run boiler and feed a steam machine while boiler is on, leave the world and then join the world again, the pipes will display they are connected but the machine will run out of steam and the boiler will loose water though it is hooked up. 

Same thing happens with wires/cables.

**Comment by 34486** [comment]
My bad on wording that, I am feeing water to steam boiler using wood pipes, then out of steam boiler is steam in copper pipes. I can also run a steam turbine and use tin wire to a batbox, if I unload the world and load the world again all of them stop working. The boiler will produce steam, but the pipes wont carry it, it will consume water, but will not get water because the wooden pipes are not working. The turbine generator will shut off due to no steam. Same thing seems to happen with any sort of wire/pipe.

**Comment by lukaslcf** [comment]
I tried to replicate what you mentioned, and tried to reload the world, but I could not replicate the issue you mentioned. Wires are still connected to each other.

![image](https://github.com/GregTechCEu/GregTechCEu-1.19/assets/3573777/e368debb-687f-4523-9d34-76ff8cd67d23)

FYI I am testing on the newest commit on the `master` branch.

**Comment by lukaslcf** [comment]
> It can do one operation without letting steam out. But doing a stack is indeed probably a bug...
> 
> I still don't see the "disconnect" that you mention though

I will open a separate issue for "processing stack despite vent blocked"

**Comment by 34486** [comment]
I guess thats the one I am using, I got the build from the latest autobuild. They will display that they are connected but are they transferring steam water and electricity after the world reload? When I was looking at them attempting to record a video I now notice that in my Jade mod display, that the internal storage for the pipes is always displayed empty, now once the world is reloaded and the pipes stop working, they dont display that they have any internal storage. I will set up another test environment with JUST gtceu and see what I can figure out.

**Comment by lukaslcf** [comment]
For me the pipes also display themselves as emtpy, but water, steam and electricity are still being moved through them.

**Comment by 34486** [comment]
I am now using 43.2.0 and no change. Same issue. Gregtech is 1.0.4 from the latest auto build, I assume we are on the same version/build?

**Comment by lukaslcf** [comment]
Thanks for the patience in going through this issue with me :)

I think we are using different builds. I am building mine locally by cloning the repo and build it through gradle. The autobuild one is probably broken, you might need to wait for the next build then, or you can try building it yourself.

**Comment by 34486** [comment]
Just used latest auto build, same issues again?!?!?!? I guess the latest auto build is behind what you are compiling still?

**Comment by lukaslcf** [comment]
Probably, as it is built on another branch.

Afaik, the autobuild feature is still being implemented.

**Comment by Yefancy** [comment]
@34486 
could you help test this build? whether the pipe issues fixed
[pipe_pr.zip](https://github.com/GregTechCEu/GregTechCEu-1.19/files/11844384/pipe_pr.zip)

**Comment by Yefancy** [comment]
I suppose it got fixed in the latest build

**Comment by 34486** [comment]
I am having the same issue still with the latest autobuild. I see the way you connect pipes is different now (you manually do it with a wrench). I disconnect all pipes manually with a wrench and reconnect them and it still will not flow/transfer. I must break and replace pipes. I am having a crash issue with wires I will have to play with that some more, will probably be its own bug. I will also try your pipes_pr.zip too just incase something changed from that to autobuild.

**Comment by lukaslcf** [comment]
Can you list the mods that you are using with their versions?

**Comment by 34486** [comment]
[modlist from crash.txt](https://github.com/GregTechCEu/GregTechCEu-1.19/files/11853816/modlist.from.crash.txt)
 
I will try without mods on other computer when I get it going here (hopefully soon).

**Comment by lukaslcf** [comment]
Just like to note that the dev env version of architectury is 3 major version older than the one you use.

**Comment by 34486** [comment]
Same issue with nothing but gt 1.0.5_16 installed.

**Comment by 34486** [comment]
Officially fixed, thank you great dev team!!!!!! 👍

**Comment by TechLord22** [comment]
Was fixed in 3ea758e, for reference.
